nav2-usability/scenario-code/dynamic-linking/dynamic_linking_vrouter.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Dynamic linking example
+/// Done using VRouter
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:vrouter/vrouter.dart';
+
+void main() {
+  runApp(WishlistApp());
+}
+
+class Wishlist {
+  final String id;
+
+  Wishlist(this.id);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Wishlist> wishlists = <Wishlist>[];
+
+  void addWishlist(Wishlist wishlist) {
+    wishlists.add(wishlist);
+    notifyListeners();
+  }
+}
+
+class WishlistApp extends StatefulWidget {
+  @override
+  _WishlistAppState createState() => _WishlistAppState();
+}
+
+class _WishlistAppState extends State<WishlistApp> {
+  final AppState appState = AppState();
+  final vRouterKey = GlobalKey<VRouterState>();
+
+  @override
+  Widget build(BuildContext context) {
+    return VRouter(
+      routes: [
+        VWidget(
+          path: '/',
+          widget: WishlistListScreen(wishlists: appState.wishlists, onCreate: onCreate),
+        ),
+        VWidget(
+          path: r'/wishlist/:id(\d+)',
+          widget: WishlistScreen(
+              wishlist: appState.wishlists.firstWhere(
+                    (element) => element.id == vRouterKey.currentState!.pathParameters['id'],
+              )),
+        ),
+      ],
+    );
+  }
+
+  void onCreate(String value) {
+    final wishlist = Wishlist(value);
+    appState.addWishlist(wishlist);
+    vRouterKey.currentState!.push('/wishlist/$value');
+  }
+}
+
+class WishlistListScreen extends StatelessWidget {
+  final List<Wishlist> wishlists;
+  final ValueChanged<String> onCreate;
+
+  WishlistListScreen({
+    required this.wishlists,
+    required this.onCreate,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: ListView(
+        children: [
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Text('Navigate to /wishlist/<ID> in the URL bar to dynamically '
+                'create a new wishlist.'),
+          ),
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: ElevatedButton(
+              onPressed: () {
+                var randomInt = Random().nextInt(10000);
+                onCreate('$randomInt');
+              },
+              child: Text('Create a new Wishlist'),
+            ),
+          ),
+          for (var i = 0; i < wishlists.length; i++)
+            ListTile(
+              title: Text('Wishlist ${i + 1}'),
+              subtitle: Text(wishlists[i].id),
+              onTap: () =>
+                  context.vRouter.push('/wishlist/${wishlists[i].id}');,
```

Comment:
```suggestion
                  context.vRouter.push('/wishlist/${wishlists[i].id}'),
```

Reply:
Nice catch, implemented directly in https://github.com/flutter/uxr/pull/28/commits/02fb450fd3d4774214dae457ae76e472b19fba85